sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_schema_registry_client.py

```diff
@@ -114,16 +114,18 @@ def register_schema(
         except AttributeError:
             pass
 
+        http_request_keywords = ["params", "headers", "json", "data", "files"]
+        http_request_kwargs = {key: kwargs.pop(key, None) for key in http_request_keywords if key in kwargs}
```

Comment:
This is a little repetitive so I can move it out to a helper method in utils.

Reply:
yes, let's move it to helper.
also I think we could put `"content_type" into the `http_request_keywords` as well, so that we don't need the extra `content_type=kwargs.pop("content_type", "application/json")` when building request

Reply:
quick clarification: do you mean we should do this `content_type=kwargs.pop("content_type", "application/json")` in the util method? or that we should do the following? so that content_type is set to None by default unless included? like so:
`
http_request_keywords = ["content_type", "params", "headers", "json", "data", "files"]
http_request_kwargs = {key: kwargs.pop(key, None) for key in http_request_keywords if key in kwargs}
`

Reply:
I'm not sure, in either case, that this would work b/c `get_schema` does not take `content_type` as a kwarg to the build_request method.

Reply:
oh, sorry for not making it clear, depending on the current implementation, I think it could be done in the _schema_registry_client.py (not util.py) like the following:
```python
        http_request_kwargs = get_http_request_kwargs(content_type=kwargs.pop("content_type", "application/json"), kwargs)
        request = schema_rest.build_register_request(
            group_name=group_name,
            schema_name=name,
            content=schema_definition,
            serialization_type=format,
            **http_request_kwargs
        )
```

Reply:
so according the the HttpRequest class that Izzy mentioned, `content_type` is not a keyword: https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/azure/core/rest/_rest_py3.py#L91

so I don't think the above would work?

Reply:
the above would technically work as build_get_by_id_request would pop the `content_type`.
you're right from design perspective, `HttpRequest` doesn't take `content_type` (content_type is part of the header).

but I feel a bit strange about the usage of the keyword arguments.. maybe it's because `content_type` is not documented in the docstring..

let's just keep the implementation for now